src/client/ua_client.c

```diff
@@ -20,6 +21,15 @@
 static void UA_Client_init(UA_Client* client, UA_ClientConfig config) {
     memset(client, 0, sizeof(UA_Client));
     client->channel.connection = &client->connection;
+    client->channel.endpoint = (UA_Endpoint*)UA_malloc(sizeof(UA_Endpoint));
```

Comment:
The _init function might be called several times internally via _reset.
Maybe Client_new is the right place for the allocation.
Note that the pointer will be NULLed in a subsequent _reset.